crates/typst/src/doc.rs

```diff
@@ -668,6 +668,8 @@ pub enum Meta {
     Elem(Content),
     /// The numbering of the current page.
     PageNumbering(Value),
+    /// The page label of the current page.
+    PageLabel(NonZeroUsize, Value),
```

Comment:
since LogicalNumbering is a PDF-only thing, I think it would be reasonable to move it into typst (especially since the PDF exporter depends on it having a specific structure, which is not the case for `PageNumbering`, that's just pass-through). that would lead to a lot more type safety.